client/bindings/wasm/src/message_handler.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::rc::Rc;
+
+use iota_client::message_interface::{create_message_handler, ClientMessageHandler, Message, Response};
+use tokio::sync::mpsc::unbounded_channel;
+use wasm_bindgen::{prelude::*, JsCast};
+use wasm_bindgen_futures::future_to_promise;
+
+/// The Client message handler.
+#[wasm_bindgen(js_name = MessageHandler)]
+pub struct WasmMessageHandler {
+    handler: Rc<ClientMessageHandler>,
+}
+
+/// Creates a message handler with the given client options.
+#[wasm_bindgen(js_name = messageHandlerNew)]
+#[allow(non_snake_case)]
```

Comment:
Is this personal preference or is camelCase required for the bindings somehow ?

Reply:
It is required because it overwrites the function exported by your nodejs bindings of the same name (also in camelCase).

Reply:
`non_snake_case` here refers to the `clientOptions` parameter in the next line. We use that because that parameter ends up in the docs and JS convention is camelCase rather than snake_case.

Reply:
Makes sense 👍🏻